app/toolbox.cpp

```diff
@@ -35,130 +36,102 @@ QString GetToolTips(QString strCommandName)
-    mClearButton->setWhatsThis(tr("Clear Tool (%1)")
-                               .arg(GetToolTips(CMD_CLEAR_FRAME)));
-    mSmudgeButton->setWhatsThis(tr("Smudge Tool (%1)")
-                                .arg(GetToolTips(CMD_TOOL_SMUDGE)));
-
-    mPencilButton->setCheckable(true);
-    mPenButton->setCheckable(true);
-    mPolylineButton->setCheckable(true);
-    mBucketButton->setCheckable(true);
-    mBrushButton->setCheckable(true);
-    mSmudgeButton->setCheckable(true);
-    mEyedropperButton->setCheckable(true);
-    mSelectButton->setCheckable(true);
-    mMoveButton->setCheckable(true);
-    mHandButton->setCheckable(true);
-    mEraserButton->setCheckable(true);
-    mPencilButton->setChecked(true);
-
-    mGridLayout->setMargin(0);
-    mGridLayout->setSpacing(0);
-
-    QWidget* toolGroup = new QWidget(this);
-
-    setWidget(toolGroup);
-    toolGroup->setLayout(mGridLayout);
-
-    connect(mPencilButton, &QToolButton::clicked, this, &ToolBoxWidget::pencilOn);
-    connect(mEraserButton, &QToolButton::clicked, this, &ToolBoxWidget::eraserOn);
-    connect(mSelectButton, &QToolButton::clicked, this, &ToolBoxWidget::selectOn);
-    connect(mMoveButton, &QToolButton::clicked, this, &ToolBoxWidget::moveOn);
-    connect(mPenButton, &QToolButton::clicked, this, &ToolBoxWidget::penOn);
-    connect(mHandButton, &QToolButton::clicked, this, &ToolBoxWidget::handOn);
-    connect(mPolylineButton, &QToolButton::clicked, this, &ToolBoxWidget::polylineOn);
-    connect(mBucketButton, &QToolButton::clicked, this, &ToolBoxWidget::bucketOn);
-    connect(mEyedropperButton, &QToolButton::clicked, this, &ToolBoxWidget::eyedropperOn);
-    connect(mBrushButton, &QToolButton::clicked, this, &ToolBoxWidget::brushOn);
-    connect(mSmudgeButton, &QToolButton::clicked, this, &ToolBoxWidget::smudgeOn);
-    connect(mClearButton, &QToolButton::clicked, this, &ToolBoxWidget::clearButtonClicked);
-
-    connect(this, &QDockWidget::dockLocationChanged, this, &ToolBoxWidget::getDockLocation);
+#ifdef __APPLE__
+    // Only Mac needs this. ToolButton is naturally borderless on Win/Linux.
+    QString sStyle =
+        "QToolButton { border: 0px; }"
+        "QToolButton:pressed { border: 1px solid #ADADAD; border-radius: 2px; background-color: #D5D5D5; }"
+        "QToolButton:checked { border: 1px solid #ADADAD; border-radius: 2px; background-color: #D5D5D5; }";
+    ui->pencilButton->setStyleSheet(sStyle);
+    ui->selectButton->setStyleSheet(sStyle);
+    ui->moveButton->setStyleSheet(sStyle);
+    ui->handButton->setStyleSheet(sStyle);
+    ui->penButton->setStyleSheet(sStyle);
+    ui->eraserButton->setStyleSheet(sStyle);
+    ui->polylineButton->setStyleSheet(sStyle);
+    ui->bucketButton->setStyleSheet(sStyle);
+    ui->brushButton->setStyleSheet(sStyle);
+    ui->eyedropperButton->setStyleSheet(sStyle);
+    ui->clearButton->setStyleSheet(sStyle);
+    ui->smudgeButton->setStyleSheet(sStyle);
```

Comment:
Setting the stylesheet in the parent widget will work, seems it doesn't need to repeat so many times.

Reply:
Yes, but the designer can’t set it only on OS X I believe.